# Mapping Omics datasets on KEGG Metabolic Pathways

##
# KEGGCharter: Enhancing Omics Data Visualization on KEGG Pathways
  
# Context and Motivation
<div align="justify">
KEGG (Kyoto Encyclopedia of Genes and Genomes) is a comprehensive database resource that integrates genomic, chemical, and systemic functional information. Specifically, KEGG metabolic maps provide a visual representation of metabolic pathways, offering insights into the complex biological systems. KEGGCharter emerges as a tool designed to superimpose differential gene expression and taxonomy information onto KEGG metabolic maps, utilizing Biopython for manipulation and visualization. Presently, KEGGCharter depicts differential gene expression and gene taxonomic assignment as separate entities on the maps. This separation limits the utility in directly associating microbial identities with gene expression data within a unified visual context.
</div>

# Solution and Objectives
<div align="justify">
This project aims to transform KEGGCharter outputs into multi-faceted, interactive representations that amalgamate both gene expression and taxonomic data within a single image. By incorporating Javascript objects for interactivity—inspired by the exploratory nature of Krona plots—these enhanced visualizations will allow users to delve into multiple data layers directly from the maps. The envisioned interactive elements will enable:

    Direct extraction of numeric values from the visualizations.
    Inclusion of multiple taxonomic levels within a single display.
    Easy access to links and cross-references to pertinent online databases.

The ultimate goal is to create a more informative and user-friendly experience, facilitating a deeper understanding of the interconnectedness between microbial identity and gene expression in metabolic pathways.
Features

    Interactive Metabolic Maps: Explore gene expression levels and taxonomic assignments in a unified, interactive format.
    Enhanced Data Representation: Multi-level data visualization to encapsulate complex omics datasets comprehensively.
    Direct Links to Databases: Immediate access to external resources for extended research and verification.
    User-friendly Interface: Designed with the user in mind, ensuring ease of navigation and data interpretation.
</div>

# Getting Started

[Instructions on how to install, configure, and use KEGGCharter, including any prerequisites.]

# Usage

[Step-by-step guide on how to generate interactive KEGG metabolic maps, with examples.]

# Contributing

[Information on how others can contribute to the project, including coding standards, pull request guidelines, etc.]

# License

[License information for the project.]

# Acknowledgments

[Acknowledgment of any collaborators, funding bodies, or any third-party resources that made the project possible.]

Visit [KEGGCharter on GitHub](https://github.com/iquasere/KEGGCharter) for more information and updates.



This project was developed by student [Duarte Velho](https://github.com/duartebred) (PG53481), under the guidance of João Sequeira and Andreia Salvador, in collaboration with the Center of Biological Engineering - University of Minho (CEB-UM).
<div align="justify">
This repository was created as part of the UC Project (2023/24), of the Master's Degree in Bioinformatics at the School of Engineering of the University of Minho, and aims to bring together the scripts, notebooks and built throughout the work, as well as articles in their different versions, presentation slides and all relevant supplementary material (data, results, software developed, etc).
This repository stands out for the originality and authorship of the written code.
</div>

##
